sql/src/main/java/org/apache/druid/sql/calcite/planner/QueryUtils.java

```diff
@@ -55,4 +58,20 @@ public static List<ColumnMapping> buildColumnMappings(
 
     return columnMappings;
   }
+
+  public static JoinAlgorithm getJoinAlgorithm(Join join, PlannerContext plannerContext)
+  {
+    RelHint closestHint = null;
+    for (RelHint hint : join.getHints()) {
+      if (closestHint == null || hint.inheritPath.size() < closestHint.inheritPath.size()) {
+        closestHint = hint;
+      }
```

Comment:
this assumes that all hints are joinhints - which may not be true in the future

Reply:
Added a check here